api/src/main/java/org/openmrs/module/htmlformentryui/tag/UiIncludeTagHandler.java

```diff
@@ -82,5 +99,27 @@ public boolean doStartTag(FormEntrySession formEntrySession, PrintWriter out, No
     public void doEndTag(FormEntrySession formEntrySession, PrintWriter out, Node parent, Node node) throws BadFormDesignException {
         // do nothing
     }
-
+	
+    /**
+     * Parses a url parameter string into a {@link Map} of properties.
+     * 
+     * @should parse velocity expression for parameter values
+     * @return Mapped properties
+     */
+	protected Map<String, Object> parseUrlParams(String input, FormEntrySession formEntrySession) {
```

Comment:
Again, strictly speaking you shouldn't need to parse URL/URI params, Apache has all you need for that.

Velocity can lookup all aliases of a given string, whatever the string, so it can be used _before_ parsing the parameters.